### DISCUSSION TITLE
Unable to use terminals in Alpine Linux Docker image

### DISCUSSION THREAD

**Discussion by ghost** [coder/code-server#5683] [Q&A]
I have a Docker image (used with Coder) set up with dependencies and all of the packages needed to run code-server. Other Linux distributions work fine, but on Alpine Linux I get the same error every time I open a terminal in code-server:

`The connection to the terminal's pty host process is unresponsive, the terminals may stop working.`

Along with that error, I also receive:

`Remote Extension host terminated unexpectedly 3 times within the last 5 minutes.`

I'm sure it's related to the lack of Glibc and usage of Musl and Busybox. I'm not sure if I'm just missing something here. Has anybody else dealt with this issue before?

The container is running Alpine Linux 3.16.2, AMD64.

**Comment by jsjoeio** [comment]
Maybe @code-asher ?

**Comment by code-asher** [comment]
How was code-server installed?  Built from scratch, npm, GitHub release?

**Reply by ghost** [reply]
I used the install script, which installs using npm by default on Alpine.

**Reply by code-asher** [reply]
It might make sense to run `npm` with the verbose flag to see if it shows any errors in its output.  My thinking is that it could be an issue with building the native pty module.

```
npm install -g code-server --unsafe-perm --dd
```

If that does not help running code-server with the verbose flag could show something:

```
code-server --verbose
```

A last-ditch thing I would try is seeing if `SHELL` is set and set it to whatever shell you have available if not.  Maybe Code is trying to spawn bash and it is not installed or something but this is a shot in the dark; I am not sure this can actually happen.
